core/services/ocr2/plugins/ocr2vrf/coordinator/coordinator.go

```diff
@@ -178,6 +185,13 @@ func (c *coordinator) ReportBlocks(
 	}
 	currentHeight := currentHead.Number
 
+	// Get mapping of block numbers to block hashes.
+	blockhashesMapping, err := c.getBlockhashesMapping(ctx, uint64(currentHeight-c.lookbackBlocks), uint64(currentHeight))
```

Comment:
Since you query for the blockhashes here, I think we can get rid of the query in `getBlocks`. Perhaps we can return a mapping from block number to `LogPollerBlock` instead, which has blockhash as part of it? Then we can pass this into `getBlocks` - in that case it would just be doing a map lookup instead of another database query.

Reply:
we actually don't need getBlocks at all, I'm replacing it with the more simple getBlockhashesMapping